scripts/resend-in-progress-conditional-alerts-and-broadcast-emails.py

```diff
@@ -0,0 +1,28 @@
+from corehq.apps.sms.models import MessagingSubEvent
+from corehq.apps.sms.models import Email
+from corehq.apps.hqwebapp.tasks import send_mail_async
+from datetime import date
+
+last_deploy_date = date(2023, 10, 18)
+
+#can optionally do one domain at a time
+subevents = MessagingSubEvent.objects.filter(
+    status='PRG',
+    date__gte=last_deploy_date
+)
```

Comment:
Will this re-send messages that were just queued moments before running this? I don't have a fully formed mental model of how this system works, but naively I would think it would.

To avoid that you could give it an end date for like 10 minutes ago—some time long enough in the past that anything just queued then will have already been sent, but recent enough that it is after the fix was deployed. Alternatively, you could run it twice 1 minute apart (long enough for any realtime messages to have made it through the queue), and only include subevents that are in both. The first method (just picking a well chosen end date) sounds simpler.

Reply:
agreed [1e26f26](https://github.com/dimagi/commcare-hq/pull/33666/commits/1e26f268751c9c4e01f8e3b7970c040c69b13587)

Reply:
Looks good